pkg/configmap/configmap_writer.go

```diff
@@ -108,6 +135,10 @@ func (c *ConfigMapWriter) Populate(maxDataSizeLimit uint64) error {
 				if err != nil {
 					return err
 				}
+				err = validateAnnotationKeys(annotationsFile)
```

Comment:
Sorry for not pointing this out earlier. The idea for validation is to check the configmap data. While it's not much different from the bundle data, it'll check a little more code.

Reply:
@jpeeler In the bug report, the invalid `annotation` yaml contains invalid keys, as a result of which the configmap is not populated the way it was expected. 
This PR checks that the valid keys are present in the `annotation` yaml, and throws an error if any expected key is missing. It does not however attempt to validate the values of the keys. 
I was under the assumption that the values are validated elsewhere, do you think we should validate that the keys are present, and that the values of those keys are of the right format here?

Reply:
Yes, I think the configmap values should be verified. It doesn't change the code all that much aside from moving this call to below the configmap call and validating the annotations instead of the file data. Put another way - when generating the bundle there will be validation there checking the format, so on this side checking the configmap is best in my opinion.

Reply:
I see your point. Updated the PR.